packages/calling/calling/src/CallingClient/CallingClient.ts

```diff
@@ -195,40 +89,75 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
     if (!this.sdkConnector.getWebex()) {
       SDKConnector.setWebex(webex);
     }
-
+    this.mutex = new Mutex();
     this.webex = this.sdkConnector.getWebex();
 
     this.sdkConfig = config;
     const serviceData = this.sdkConfig?.serviceData?.indicator
       ? this.sdkConfig.serviceData
       : {indicator: ServiceIndicator.CALLING, domain: ''};
 
+    const logLevel = this.sdkConfig?.logger?.level ? this.sdkConfig.logger.level : LOGGER.ERROR;
     this.validateServiceData(serviceData);
 
-    this.registration = createRegistration(this.webex, serviceData);
+    this.registration = createRegistration(
+      this.webex,
+      serviceData,
+      this.mutex,
+      this.callingClientEmitter,
+      logLevel
+    );
     this.callManager = getCallManager(this.webex, serviceData.indicator);
-    this.metricManager = getMetricManager(this.webex, serviceData.indicator);
 
     this.mediaEngine = Media;
+
     this.primaryMobiusUris = [];
     this.backupMobiusUris = [];
-    this.rehomingIntervalMin = DEFAULT_REHOMING_INTERVAL_MIN;
-    this.rehomingIntervalMax = DEFAULT_REHOMING_INTERVAL_MAX;
-    this.failback429RetryAttempts = 0;
-    this.clientError = createClientError('', {}, ERROR_TYPE.DEFAULT, MobiusStatus.DEFAULT);
-    this.isRegistered = MobiusStatus.DEFAULT;
-    this.reconnectPending = false;
+
+    this.registration.setRegistrationStatus(MobiusStatus.DEFAULT);
     this.registerSessionsListener();
-    const logLevel = this.sdkConfig?.logger?.level ? this.sdkConfig.logger.level : LOGGER.ERROR;
 
-    this.mutex = new Mutex();
     log.setLogger(logLevel, CALLING_CLIENT_FILE);
-    /* Better to run the timer once rather than after every registration */
-    this.detectNetworkChange();
+
     this.incomingCallListener();
     this.registerCallsClearedListener();
   }
 
+  private callingClientEmitter = (
+    event: EVENT_KEYS,
+    deviceInfo?: IDeviceInfo,
+    clientError?: CallingClientError
+  ) => {
+    switch (event) {
+      case EVENT_KEYS.REGISTERED:
+        if (deviceInfo) {
```

Comment:
I don't see any case where Registered event will be dispatched and device info won't be present then why was this condition required ? Is this because of it being optional in the arguments ?

Reply:
> Is this because of it being optional in the arguments 
Yes.
It is optional since, UNREGISTERED, RECONNECTED & RECONNECTING doesn't need the argument.